lib/logical/ElementDefinition.js

```diff
@@ -640,6 +640,37 @@ class ElementDefinition {
     return this;
   }
 
+  /**
+   * Fixes a fixed value to the element or a child of the element indicated by the path.  If the target element is
+   * in a deeper path than currently represented in the structure definition, parent level elements will be "expanded"
+   * (or "unrolled") based on their definition.  This means that calling this function may result in increasing the
+   * number of elements in the structure definition.
+   * @param {boolean} value - the fixed value to fix
+   * @param {string} [path] - if the fixed should be fixed to a child element, the dot-separated path of that element
+   * @param {function({code: string, profile?: string[], targetProfile?: string[], aggregation?: string[], versioning?: string}):StructureDefinition} [resolve] - a function that can resolve a type to a StructureDefinition instance
+   * @returns {ElementDefinition} the element to which the value was fixed (may be a child of the current element)
+   */
+  fixFixedValue(value, path, resolve) {
+    if (path && path.length > 0) {
+      const child = this.findChild(path, resolve);
+      if (child) {
+        return child.fixFixedValue(value, null, resolve);
+      }
+      return;
+    }
+
+    // This is the element to fix it to
+    // Get the type that we are fixing and capitalize the first letter
+    const lowercaseType = this.type[0].code;
```

Comment:
This assumes that the first type is the type we want to fix.  Is this a safe assumption?  Have you tested this with an element that is a choice (for example `string or integer`) -- and ensuring that if you fix the string, you get `fixedString` and if you fix the integer you get `fixedInteger`?

Reply:
```
Entry:      Test1
Property:   ChoiceTest 0..1
            ChoiceTest[uri] = "http://example.org"

Entry:      Test2
Property:   ChoiceTest 0..1
            ChoiceTest[boolean] = true

Entry:      Test3
Property:   ChoiceTest 0..1
            ChoiceTest[integer] = 5

Element:    ChoiceTest
Value:      string or integer or uri or date or Boolean
```
I have tested on a case like this, and you do get `fixedInteger` or `fixedUri` where you would expect. In all cases I could find, that type array always ends up being of length 1. The first part of the function burrows down into the child that has been selected using the brackets. And this child has a known type.

Reply:
Great.  I couldn't remember exactly how it worked -- and if it returned the element w/ the full choice (`type.length > 1`) or if it created the specific element for the requested choice.